src/transformers/models/table_transformer/convert_table_transformer_to_hf_no_timm.py

```diff
@@ -402,34 +423,29 @@ def convert_table_transformer_checkpoint(checkpoint_url, pytorch_dump_folder_pat
 
     if push_to_hub:
         # Push model to HF hub
-        logger.info("Pushing model to the hub...")
-        model_name = (
-            "microsoft/table-transformer-detection"
-            if "detection" in checkpoint_url
-            else "microsoft/table-transformer-structure-recognition"
-        )
-        model.push_to_hub(model_name, revision="no_timm")
-        image_processor.push_to_hub(model_name, revision="no_timm")
+        logger.info("Pushing model and image processor to the hub...")
+        model.push_to_hub(f"microsoft/{model_name}", revision="no_timm")
+        image_processor.push_to_hub(f"microsoft/{model_name}", revision="no_timm")
 
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
 
     parser.add_argument(
-        "--checkpoint_url",
-        default="https://pubtables1m.blob.core.windows.net/model/pubtables1m_detection_detr_r18.pth",
+        "--model_name",
+        default="table-transformer-detection",
         type=str,
```

Comment:
Looks good to me but not BC, we can try to parse the url to keep the previous behaviour / allow converting other raw urls